### ISSUE TITLE
geometry_objects read/write

### ISSUE THREAD

**Issue by JakubVaverka83** [gprMax/gprMax#466]
Hello,

I have a question related to inserting geometry into a model. It looks that I didn't understand it or something works not right.

My idea was to generate h5 file by **geometry_objects_write** modify it by MatLab and read it back by **geometry_objects_read**.

I have rewrote /data variable in the h5 file, but the result was extremely suspicious. 

H5 file looked as expected, geometry generated by **geometry_view** looked as expected but the result didn't correspond to it. 

Than I realize that the geometry is also written in /ID variable in h5 file generated by *geometry_objects_write**.  **geometry_view** use /data variable for visualization but simulation use /ID variable. 

Simulation didn't correspond to geometry (**geometry_view**) when I rewrite only /data file. This is extremely confusing. 
It is necessary to rewrite also /ID variable in h5 file to obtain result as expected.  

Should it really work like this? There is no information about /ID variable in documentation and a note that it is necessary to rewrite it as well.

Is there something I didn't understand or did wrong?

To generate h5 file by **geometry_objects_write**, modify it (/data variable) and import back by **geometry_objects_read** looked as simple and straightforward way how to import complicated geometry. I suppose that I am not the first one to do it this way.

Thank you 

Jakub Vaverka

**Comment by Aryan022mittal** [comment]
hie, 
can i solve this issue.

**Comment by Aryan022mittal** [comment]
allow me to help you with this issue

**Comment by JakubVaverka83** [comment]
Thank You. I have solved the problem by modifying both /data and /ID arrays (I have no idea if this is the best solution). My main point was that more people trying to generate h5 file by **geometry_objects_write** and then to modifi it can have the same problem (even without noticing it). It should be probably good idea to mentioned it in the documentation that this is not a good aproach or mention that /ID array is crucial for the simulation (not /data array).

**Comment by craig-warren** [comment]
@JakubVaverka83 @Aryan022mittal Yes - users should not modify the geometry that is exported from gprMax using #geometry_objects_write. As you have spotted when geometry created by gprMax is exported there are additional arrays written to file that gprMax uses internally, for things like dielectric smoothing. If you edit these yourself then you risk corrupting your model. This functionality is intended to speed up the reading of geometry already generated by gprMax. Creating geometry outside gprMax and then using #geometry_objects_read is a better approach here. However, I will add a clearer warning in the docs.

**Comment by JakubVaverka83** [comment]
Thank you. The warning in the docs could save time for others trying similar approach as i did. I was just confused as this way looked more straightforward than creating geometry outside from the scratch.
